DESIGN.md

```diff
@@ -0,0 +1,102 @@
+
+
+## Client
+
+Client is a simple CLI for the communication with the server. It will allow running a set of commands against the server:
+
+1. Start a job: passes a command provided by the user for direct execution on the server side. The ID of the job will be sent in response.
+If the user wants to limit resources that will be available to a job upon execution he is required to do it while sending a start request.
+No update functionality will be provided in terms of resources control - if the user wants to add or change some limits he is required to stop the job and create a new one with the required parameters.
+The following set of flags is used for this purpose, user can set the required limit in one of the groups:
+    * **mem** - memory limit for a job in megabytes
+    * **cpu** - cpu share in percents (1-100) available to this job
+    * **io** - proportion of I/O access (1-100) available to this job
+1. Stop the job: a user is required to provide a job ID for the job termination. There are going to be 3 modes of this command:
+    * In default mode (when only job ID is provided) _SIGTERM_ is used to terminate a job. 
+    * If the user provides a **force** flag it triggers _KILL_ to be sent for the command termination.
+    It can be useful if, for example, it takes too long for a command to finish in normal mode, if the user wants to be sure the command terminates as soon as possible or if it is not possible to terminate it otherwise for some reason.
+    * If the user provides a **gentle** flag it triggers _SIGINT_ to be sent.
+    It might be essential to finish some commands in these modes in order to trigger whatever functionality they might have set up to be executed on the interruption request.
+1. Get the status of the job. Requires only the job ID to be sent, the user gets in return the job status, all the job resource limits set upon job creation and exit code (applies only if the job is in the finished or stopped status)
+1. Stream the output of the job. Requires only the job ID, starts the stream of the job stdout - the user gets everything that was written by the command until that moment and continues to get the command logs in real time until either the job is finished/terminated or the user interrupts the stream command execution (CTRL-C).
+It is a completely valid scenario to request the logs of both stdout and stderr (or even stdout and once again stdout) of the same job at the same time. 
+    * **err** - optional flag, if provided starts the stream of stderr instead of stdout
+
+Some examples can be found under the Usage section of the readme file.
+
+
+## Security
+
+Client and server keys and certificates will be either presented in the repository or generated using the provided script. 
+
+mTLS authentication will be used. As we don't want to introduce some sophisticated auth system in this exercise we might consider all the users with valid certificates to be authorized within the system. 
```

Comment:
Could you expand on this a bit? Describe which secrets will be generated and how client/server will use them please.

Reply:
Added more info above, please ping me if I'm still missing something or this is not enough